src/components/pages/doc.js

```diff
@@ -125,6 +127,62 @@ function Doc({ initialState, PageTemplate }) {
     hasPendingTemplateSectionKeys: !!pendingTemplateSectionKeys.length
   }));
 
+  const findSectionKeyToScrollTo = () => {
+    if (!window.scrollY) {
+      return null;
+    }
+
+    const measurements = [...window.document.body.querySelectorAll('[data-section-key]')]
+      .map(element => ({ element, ...getViewportMeasurementsForElement(element) }));
+
+    if (!measurements.length) {
+      return null;
+    }
+
+    if (measurements.length === 1) {
+      return measurements[0].element.getAttribute('data-section-key');
+    }
+
+    const firstElementStartingInViewport = measurements.find(m => m.elementTopIsInViewport);
+    if (firstElementStartingInViewport) {
+      return firstElementStartingInViewport.element.getAttribute('data-section-key');
+    }
+
+    const elementCoveringMostViewport = measurements.sort(by(x => x.viewportCoverage, 'desc'))[0];
+    if (elementCoveringMostViewport) {
+      return elementCoveringMostViewport.element.getAttribute('data-section-key');
+    }
+
+    return null;
+  };
+
+  const tryBringSectionIntoView = sectionKey => {
+    const element = window.document.querySelector(`[data-section-key="${sectionKey}"]`);
+    if (!element) {
+      return;
+    }
+
+    const measurement = getViewportMeasurementsForElement(element);
+    if (!measurement.elementTopIsInViewport) {
+      element.scrollIntoView({ behavior: 'smooth' });
+    }
+  };
```

Comment:
Suggestion from me would be to move `findSectionKeyToScrollTo` and `tryBringSectionIntoView` into `doc-utils.js` to reduce the logic on this huge page a big.

Reply:
Good idea. Done! It also helped to keep the custom data attribute names in one place :+1: